commands/admin/status.ts

```diff
@@ -9,67 +9,87 @@ exports.usage = 'status {STATUS} {ICON}'
 exports.example = `status Bitte nicht stören -dnd\n
 status `
 
-const presence = {
+/**
+ * Presence object to use for custom presences.
+ */
+const presence: setPresenceData = {
   activities: [
     {
       name: '',
       type: 'PLAYING',
     },
   ],
-  status: '',
+  status: 'online',
 }
 
 exports.presence = presence
 
 exports.run = (client: DiscordClient, message: Message) => {
+  /**
+   * Check if user has the correct rights to execute the command.
+   */
   if (!Object.values(client.config.ids.acceptedAdmins).includes(message.author.id)) {
     return client.reply(message, { content: 'You do not have the permissions to perform that command.' })
   }
 
+  /**
+   * Content of message.
+   */
   let messageContent: string = message.content
 
   messageContent = messageContent.split('.status')[1]
 
-  const activityName: string = messageContent.split('-')[0]
+  /**
+   * Set activity name.
+   */
+  presence.activities[0].name = messageContent.split('-')[0]
 
-  presence.activities[0].name = activityName
+  /**
+   * Set presence status.
+   */
+  presence.status = getPresenceStatusData(messageContent)
 
-  getIcon(messageContent)
-
-  if (presence.activities[0].name === ' ' || presence.activities[0].name === '') {
-    const defaultPresence = client.config.presence[0]
-    return Presence(client, message, defaultPresence)
-  } else {
-    return Presence(client, message, presence)
-  }
+  /**
+   * Only set custom presence if defined.
+   * Otherwise use default presence from config.
+   */
+  return presence.activities[0].name === ' ' || presence.activities[0].name === ''
+    ? setPresenceData(client, message, client.config.presence[0])
+    : setPresenceData(client, message, presence)
 }
 
-function getIcon(messageContent: string): void {
+/**
+ * Fetches {@link getPresenceStatusData}
+ * @param {string} messageContent {@link Message.content}
+ * @returns {getPresenceStatusData} {@link getPresenceStatusData}
+ */
+function getPresenceStatusData(messageContent: string): getPresenceStatusData {
```

Comment:
The return type is function name? 
Need to do more investigation there...

Reply:
Imported `PresenceStatusData` as `getPresenceStatusData`.

Fixed in ba389b82aea4f3be266886666c57d0ff19630706